### PR TITLE
Add Git to environment.yml

### PR DESCRIPTION
**Pull request by MattF-NSIDC**
The version of Git that comes with Ubuntu 18 is pretty dated. My main concern is the lack of the modern [switch](https://git-scm.com/docs/git-switch) and [restore](https://git-scm.com/docs/git-restore) subcommands, which provide a more intuitive alternative to the `checkout` interface (I believe `git checkout` is unintuitive because it can both switch branches and restore files from another ref depending on invocation pattern). To avoid confusion, I prefer to avoid introducing new Git users to the `checkout` subcommand.

My other concern is the [discovery of vulnerabilities in Git](https://github.blog/2023-01-17-git-security-vulnerabilities-announced-2/) this past January. Based on [this announcement](https://lore.kernel.org/git/xmqq7cxl9h0i.fsf@gitster.g/T/#u), the fix was backported only to versions >= 2.30, whereas this image currently has 2.17.1.

This change is installing Git from conda-forge, but normally I'd use the [`git-core` ppa](https://launchpad.net/~git-core/+archive/ubuntu/ppa) for this. I [don't think there's a way](https://github.com/jupyterhub/repo2docker/issues/402) to do this without falling back to a custom Dockerfile.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -114,6 +114,9 @@ dependencies:
   - pyopencl
   # needed for Linux desktop to work
   - websockify>=0.10
+  # Includes new subcommands (`switch` & `restore`) and vulnerability fixes
+  # over the version available via ubuntu sources
+  - git ~=2.39
```
Comment (weiji14) [inline environment.yml:119]: Can we do greater or equal than?
```suggestion
  - git >=2.39
```

**Review by weiji14 — COMMENTED** [review]
Yes, I'm a big fan of `git switch` as well! Could you write a comment below with just `/condalock` to regenerate the lockfile? Thanks!

File: environment.yml (hunk 1 above)
Comment (MattF-NSIDC) [reply environment.yml:119]: I'd be happy to make this change. I have a few reasons I prefer this style and opened an issue (#46) to discuss if it was OK to use it. Can you let me know what you think? Or, I can accept this change now and we can deal with that idea later. Please let me know :)

Comment (weiji14) [reply environment.yml:119]: For consistency with the rest of the `environment.yml` file, it'll be better to use `>=` first, and then we can decide in #46 on whether to use `~=` instead. The contents of the lockfile shouldn't be affected by the choice of `>=` or `~=` in the short term.

Comment (MattF-NSIDC) [reply environment.yml:119]: :+1: sounds good!

**Comment by MattF-NSIDC** [comment]
@weiji14 I accepted your suggestion, but after doing so the commit generated by GitHub shows up as "Partially Verified". Any idea how to get around this?

**Comment by weiji14** [comment]
> @weiji14 I accepted your suggestion, but after doing so the commit generated by GitHub shows up as "Partially Verified". Any idea how to get around this?
> 
> > We cannot verify signatures from co-authors, and some of the co-authors attributed to this commit require their commits to be signed.

Only way is if you can get my signing key :stuck_out_tongue_closed_eyes: I've got vigilant mode enabled (see https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits#about-vigilant-mode), so any commits people do on my behalf will show up like that, which is fine.

**Review by weiji14 — APPROVED** [review]

**Comment by weiji14** [comment]
Thanks @MattF-NSIDC! Just a note that the dependency updates won't show up on the CryoCloud JupyterHub immediately. I think @tsnow03 will need to update some configuration following https://github.com/CryoInTheCloud/hub-image#updating-the-cryocloud-jupyterhub-to-use-a-new-image to point to the newly built docker image.

**Comment by MattF-NSIDC** [comment]
Awesome, thanks for your help getting this change out! :heart:

**Pull request merged**